### PR TITLE
DYN-7531 Catch WebView2 initialization exception

### PR DESCRIPTION
**Pull request by QilongTang**
### Purpose

Per CER report https://cer.autodesk.com/search/v2/buckets/52352002, add some code to prevent WebView2 initialization (Dynamo Home) crashing Dynamo.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

Add some code to prevent WebView2 initialization (Dynamo Home) crashing Dynamo

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 1]
```diff
@@ -1427,21 +1427,28 @@ private void LoadHomePage()
...
-                BindingOperations.SetBinding(homePage, UIElement.VisibilityProperty, visibilityBinding);
+                    BindingOperations.SetBinding(homePage, UIElement.VisibilityProperty, visibilityBinding);
 
-                this.newHomePageContainer.Children.Add(homePage);
+                    this.newHomePageContainer.Children.Add(homePage);
+                }
+                catch (Exception ex)
+                {
+                    Log(ex.Message);
+                }
```
Comment (pinzart90) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1451]: any idea what exactly is crashing?

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 2]
```diff
@@ -1427,21 +1427,28 @@ private void LoadHomePage()
         {
             if (homePage == null && (startPage != null))
             {
-                homePage = new UI.Views.HomePage();
```
Comment (pinzart90) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1430]: if we handle the crash, will Dynamo be able to continue running ok?

Comment (pinzart90) [reply src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1430]: If the home page is not shown, can Dynamo still be used ?

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs (hunk 1 above)
Comment (RobertGlobant20) [reply src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1451]: Agree with @pinzart90 we should know why/what is crashing.
I have seen several times that when opening the latest Dynamo (3.4 or 3.3) the HomePage is NOT being shown and instead the old one is shown but I didn't know exactly why (although is not always happening).

Comment (QilongTang) [reply src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:1451]: @RobertGlobant20 @pinzart90 I have removed the feature flag checking code for DynamoHome in 3.4 release, in that case, Dynamo home would always be the default. In the case it was not initialized correctly, then we display the legacy home. Adding such exception logging would help us understand why it was not working

**Comment by QilongTang** [comment]
Another thing to note here is that all of these CER reports are from 3.2.2 release, so I am not sure if the crash is still reproducible in latest Dynamo.

**Comment by QilongTang** [comment]
Re-running the self CI on https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/16866/

**Review by reddyashish — APPROVED** [review]

**Pull request merged**